lib/client.js

```diff
@@ -1942,6 +1942,16 @@ function doInitialSync(client, historyLen, includeArchived) {
                     room.addReceipt(receipts[j]);
                 }
 
+                var privateUserData = data.rooms[i].private_user_data || [];
+                for (j = 0; j < privateUserData.length; j++) {
+                    var event = _PojoToMatrixEventMapper(client)(privateUserData[j]);
```

Comment:
I think this is very subjective personally, and I'm not sure I agree - the added boilerplate and two loops and mix of map & iteration feels minging to me.  But i've done it anyway.